cmd/soroban-cli/src/commands/network/container/stop.rs

```diff
@@ -1,28 +1,53 @@
 use crate::commands::network::container::shared::{
-    connect_to_docker, Error as ConnectionError, Network, DOCKER_HOST_HELP,
+    connect_to_docker, Error as BollardConnectionError,
 };
 
+use super::shared::Args;
+
 #[derive(thiserror::Error, Debug)]
 pub enum Error {
-    #[error("Failed to stop container: {0}")]
-    StopContainerError(#[from] ConnectionError),
+    #[error("⛔ Failed to connect to docker: {0}")]
+    DockerConnectionFailed(#[from] BollardConnectionError),
+
+    #[error("⛔ Container {container_name} not found")]
+    ContainerNotFound {
+        container_name: String,
+        #[source]
+        source: bollard::errors::Error,
+    },
+
+    #[error("⛔ Failed to stop container: {0}")]
+    ContainerStopFailed(#[from] bollard::errors::Error),
 }
 
 #[derive(Debug, clap::Parser, Clone)]
 pub struct Cmd {
-    /// Network to stop
-    pub network: Network,
+    #[command(flatten)]
+    pub container_args: Args,
 
-    #[arg(short = 'd', long, help = DOCKER_HOST_HELP, env = "DOCKER_HOST")]
-    pub docker_host: Option<String>,
+    /// Container to stop
+    pub name: String,
 }
 
 impl Cmd {
     pub async fn run(&self) -> Result<(), Error> {
-        let container_name = format!("stellar-{}", self.network);
-        let docker = connect_to_docker(&self.docker_host).await?;
-        println!("ℹ️  Stopping container: {container_name}");
-        docker.stop_container(&container_name, None).await.unwrap();
+        let container_name = self.name.clone();
+        let docker = connect_to_docker(&self.container_args.docker_host).await?;
+        println!("ℹ️ Stopping container: {container_name}");
+        docker
+            .stop_container(&container_name, None)
+            .await
+            .map_err(|e| {
+                let msg = e.to_string();
+                if msg.contains("No such container") {
+                    Error::ContainerNotFound {
+                        container_name: container_name.clone(),
```

Comment:
```suggestion
        let container_name = self.name;
        let docker = connect_to_docker(&self.container_args.docker_host).await?;
        println!("ℹ️ Stopping container: {container_name}");
        docker
            .stop_container(&container_name, None)
            .await
            .map_err(|e| {
                let msg = e.to_string();
                if msg.contains("No such container") {
                    Error::ContainerNotFound {
                        container_name: container_name.clone(),
```
Only need the clone in the error closure.

Reply:
This particular change is no longer relevant after reverting https://github.com/stellar/stellar-cli/commit/d21365a0404be7a37201626dd1ecf073b706e089